docs/whirlpool/docs/02-Architecture Overview/07-TokenExtensions Support.md

```diff
@@ -0,0 +1,161 @@
+
+#### `minA/B`, `maxA/B`
+Increase Liquidity: `maxA`, `maxB`
+transfer fee Included amount (will be sent from user's token account)
+
+Decrease Liquidity: `minA`, `minB`
+transfer fee Excluded amount (will be received at user's token account)
+
+### TransferHook
+In order to use TransferHook, the owner of WhirlpoolsConfig must issue a TokenBadge for that token.
+
+#### Use of remaining_accounts
+The account used for TransferHook is different for each program used by TransferHook. Therefore, they are received through `remaining_accounts`.
+
+To clarify the context of accounts passed as `remaining_accounts`, the v2 instructions receive `remaining_accounts_info` as data. It is used to classify the accounts contained in `remaining_accounts`. (e.g. The first 3 are for mintA's TransferHook, next 2 are for for mintB's TransferHook).
+
+#### Security notes
+- In solana, indirect re-entrance will be rejected. So transfer hook program cannot call Whirlpool program.
+- TransferHook program cannot modify transfer amount. It will be called AFTER token transfer processing.
+- Token-2022 program remove signer and writable flag of the source, destination and owner when calling the TransferHo[ok program even if they are passed as extra accounts. The source and destination are not updated and the owner's signature is not used unintentionally.
+
+#### TokenBadge request
+TransferHook extension is a relatively new feature in the ecosystem and it is still unknown how it will be used.
+
+It remains to be decided what kind of TransferHook tokens TokenBadge will be issued for, but it is important that they have the following characteristics
+
+- The code of the TransferHook program is publicly available.
+- Verifiable Build has been done to ensure the code and programme match.
+- Upgrade authority have been disabled (ideal, but not required)
+- Only perform processes that pose no risk to the user (e.g. logging)
+- TransferHook must not block the transfer of tokens, or that the criteria for blocking are clearly declared, fair and reasonable.
+- TransferHook must not interfere with the operation of the pool. It must not interfere with the trade, deposit, withdraw and harvest.
+- Extensions that block transactions based on the amount of tokens to be transferred are undesirable because they cause transactions to fail in ways that are surprising to pool users. On the other hand, failing transfers to sanctioned wallets according to some public standard will not obstruct pool usage for many users.
+- TransferHook must not request large numbers of accounts that would increase transaction size.
+- TransferHook must not execute any token transfers (including WSOL and native SOL).
+- TransferHook must not impose any kind of additional fees.
```

Comment:
Not sure if these last two are even possible since the signers are not passed on to the TransferHook program. Like it would only be able to come from a third account not owned by the trader/depositor